### PR TITLE
Implement resume uploads

### PR DESCRIPTION
**Pull request by shanekim28**
Addreses #311
* Added resume url column
* Added a resume upload route in the Users controller
* Created new filetype enums
* Added constants for upload path

### CODE REVIEW

**Review by shravanhariharan2 — COMMENTED** [review]
Nice work! Minor comments, should be ready to merge once the prod & testing env changes mentioned below are addressed

**Comment by shravanhariharan2** [comment]
Also, let's add some tests to make sure s3 gets passed in the right config and that non-pdf files get properly rejected

File: config/index.ts
Comment (shanekim28) [reply config/index.ts:68]: Caprover's been updated to include these environment variables. Will update .env.example on next push.

**Comment by shanekim28** [comment]
@shravanhariharan2 The upload resume route does not throw an error when the resume is too big; it allows the resume to be uploaded regardless of file size. The file options use the .env file for max resume upload file size. When comparing the actual and max file sizes, the actual is indeed larger than the max, but it still allows it to upload.

Are there no automatic file size checks when the file is uploaded to the route?

**Comment by shravanhariharan2** [comment]
> @shravanhariharan2 The upload resume route does not throw an error when the resume is too big; it allows the resume to be uploaded regardless of file size. The file options use the .env file for max resume upload file size. When comparing the actual and max file sizes, the actual is indeed larger than the max, but it still allows it to upload.
Are there no automatic file size checks when the file is uploaded to the route?

The reason why an error is not being thrown is because we are directly calling `userController.updateResume()` in our test code, which does not actually call the middleware. If you were to make an API request over Postman to this route, it would properly perform file size checks.

I'd love to say that there's an easy solution here, but to be honest we haven't figured out middleware testing in this codebase, so my inclination is to skip testing for file sizes and at least make sure it works manually, and we can discuss a middleware testing plan down the road and revisit this specific case then. What do you think?

**Comment by shravanhariharan2** [comment]
Holding off on a full review until we have the resume upload requirements fully figured out: https://discord.com/channels/573028991527550986/791028414538186772/1022205693895376906

**Comment by shanekim28** [comment]
> > @shravanhariharan2 The upload resume route does not throw an error when the resume is too big; it allows the resume to be uploaded regardless of file size. The file options use the .env file for max resume upload file size. When comparing the actual and max file sizes, the actual is indeed larger than the max, but it still allows it to upload.
> > Are there no automatic file size checks when the file is uploaded to the route?
> 
> The reason why an error is not being thrown is because we are directly calling `userController.updateResume()` in our test code, which does not actually call the middleware. If you were to make an API request over Postman to this route, it would properly perform file size checks.
> 
> I'd love to say that there's an easy solution here, but to be honest we haven't figured out middleware testing in this codebase, so my inclination is to skip testing for file sizes and at least make sure it works manually, and we can discuss a middleware testing plan down the road and revisit this specific case then. What do you think?

It feels a little dirty to leave it out, but I also think it's the best option until we have something figured out with the middleware. I'll update the tests asap

**Comment by shanekim28** [comment]
TODO:
- [ ] Store resumes in folder with name of user's UUID in AWS bucket using user's file name instead of user's UUID
- [ ] Create opted-in column (is resume visible)
- [ ] Create upload date column

File: tests/resume.test.ts
Comment (shravanhariharan2) [inline tests/resume.test.ts:41]: Can we verify that the storage mocks for deleting and uploading files properly get called here too?

**Review by shravanhariharan2 — CHANGES_REQUESTED** [review]

File: api/controllers/ResumeController.ts
Comment (shravanhariharan2) [inline api/controllers/ResumeController.ts:30]: I think instead of having a `deleteAtUrls`, we can just do a findOne query on resumes and delete that single resume. The reason being is we'd never really have the use case for deleting all other resumes for a user - for a "single resume per user" case, we just delete the previous one and upload the new one, and for a "multiple resumes per user" case, we would either just not delete any resumes, or delete single resumes within a category (e.g. if a user uploads a DEVOPS resume, and then another DEVOPS resume, we would delete the single previous DEVOPS resume.), depending on the requirements.

So I think it makes sense to keep it simple and just have a "deleteAtUrl" function that we're calling here, and not have a "deleteAtUrls" one that will always just run once.

**Review by shravanhariharan2 — COMMENTED** [review]
Great progress! One comment on our `deleteAtUrls` case, and another comment from the past review on testing that the storage mocks properly get called.

File: tests/mocks/MockFactory.ts
Comment (shravanhariharan2) [inline tests/mocks/MockFactory.ts:10]: We also need to mock the other StorageService functions here, so that no actual S3 calls get made. Right now, only `uploadToFolder` is the only fn that's mocked, but we also need to mock `upload` and `delete`.

File: models/ResumeModel.ts
Comment (shravanhariharan2) [inline models/ResumeModel.ts:16]: Should we make this default true instead of false? I'd imagine we'd only want a user to have their resume viewable by recruiters if they specify they want to, otherwise we should hide by default. Let me know if you have any opinions on this though

File: tests/data/FileFactory.ts
Comment (shravanhariharan2) [inline tests/data/FileFactory.ts:35]: nit: let's move this comment to the `pdf()` fn since that is the first occurence of these fields in the file

File: tests/resume.test.ts
Comment (shravanhariharan2) [inline tests/resume.test.ts:37]: nit: we can inline `storageSvcInstance` since we aren't using it anywhere else:
```ts
const resumeController = ControllerFactory.resume(conn, instance(storageService));
```

File: tests/resume.test.ts
Comment (shravanhariharan2) [inline tests/resume.test.ts:58]: nit: inline

File: tests/resume.test.ts
Comment (shravanhariharan2) [inline tests/resume.test.ts:80]:
```suggestion
  test('uploading resumes with the wrong filetype throws an error', async () => {
    const conn = await DatabaseConnection.get();
```

**Review by shravanhariharan2 — APPROVED** [review]
Awesome work! Love the coverage with this test suite. Last few comments, you're welcome to merge once they've all been addressed

**Pull request merged**
